.github/workflows/starknet-go-tests.yml

```diff
@@ -0,0 +1,33 @@
+name: starknet-go tests
+
+on:
+  workflow_call:
+    secrets:
+      TEST_RPC_URL:
+        required: false
+      TEST_ACCOUNT_PRIVATE_KEY:
+        required: false
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v4
+        with:
+          repository: NethermindEth/starknet.go
+          ref: main
+
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.23'
```

Comment:
Does it need to be the same as juno?  if so, It needs to be 1.23.1: https://github.com/NethermindEth/juno/blob/main/go.mod#L4

Reply:
this go version is for building starknet.go proj, not juno, so don't have to be same